### ISSUE TITLE
Incorrect pathing for games with similar names when kept locally after downloading

### ISSUE THREAD

**Issue by PimmoZz** [zach-morris/plugin.program.iarl#27]
I have set IARL to download PSX games since sometimes the download connection can be slow.
I have **Clean ROM title** disabled because I had a similor error with PSX Gran turismo 2  where the Arcade disc and de Simulation disc had the same name. I worked around this by downloading them separately and then putting them back in the correct location after downloading. All with **Clean ROM title** disabled. The Spyro games prove to be more of a challenge.

For example the Spyro games:
- Local paths:
D:\Games\Retro Games\PSX\Spyro_the_Dragon_USA\
D:\Games\Retro Games\PSX\Spyro_Year_of_the_Dragon_USA_v11\

- When both paths exists, Spyro_the_Dragon_USA does not launch, Spyro_Year_of_the_Dragon_USA_v11 does launch correctly.
- When Spyro_Year_of_the_Dragon_USA_v11 does not exist, Spyro_the_Dragon_USA does launch correctly.

The error in the debug log when trying to launch Spyro_the_Dragon_USA:
```13:49:36.926 T:13240  NOTICE: IARL:  Download and Launch started for Spyro the Dragon (USA)
13:49:36.927 T:13240   DEBUG: IARL:  Loading history cache C:\Users\pimve\AppData\Roaming\Kodi\userdata\addon_data\plugin.program.iarl\list_cache\iarl_history.pickle
13:49:36.955 T:13240   DEBUG: IARL:  List cached to file C:\Users\pimve\AppData\Roaming\Kodi\userdata\addon_data\plugin.program.iarl\list_cache\iarl_history.pickle
13:49:36.955 T:13240  NOTICE: IARL:  Download started for Spyro the Dragon (USA)
13:49:36.964 T:13240   DEBUG: IARL:  Matching files found for D:\Games\Retro Games\PSX\Spyro the Dragon (USA).zip: D:\Games\Retro Games\PSX\Spyro_Year_of_the_Dragon_USA_v11\Spyro - Year of the Dragon (USA) (v1.1).cue, D:\Games\Retro Games\PSX\Spyro_the_Dragon_USA, D:\Games\Retro Games\PSX\Spyro_the_Dragon_USA\Spyro the Dragon (USA).cue, D:\Games\Retro Games\PSX\Spyro_Year_of_the_Dragon_USA_v11\Spyro - Year of the Dragon (USA) (v1.1).bin, D:\Games\Retro Games\PSX\Spyro_the_Dragon_USA\Spyro the Dragon (USA).bin
13:49:36.964 T:13240   DEBUG: IARL:  File was found to exist locally, no overwrite option selected
13:49:36.964 T:13240   DEBUG: IARL:  Matching file that already exists: D:\Games\Retro Games\PSX\Spyro_Year_of_the_Dragon_USA_v11\Spyro - Year of the Dragon (USA) (v1.1).cue
13:49:36.964 T:13240   DEBUG: IARL:  Post DL Override command detected for Spyro the Dragon (USA) - convert_zip_m3u
13:49:36.967 T:11504   DEBUG: ------ Window Init (DialogNotification.xml) ------
13:49:36.969 T:13240   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.WindowsError'>
                                            Error Contents: (3, 'Het systeem kan het opgegeven pad niet vinden', 'D:\\Games\\Retro Games\\PSX\\Spyro_Year_of_the_Dragon_USA_v11\\Spyro_the_Dragon_USA/*.*')
                                            Traceback (most recent call last):
                                              File "C:\Users\pimve\AppData\Roaming\Kodi\addons\plugin.program.iarl\addon.py", line 1825, in onClick
                                                download_and_launch_rom(self,iarl_data)
                                              File "C:\Users\pimve\AppData\Roaming\Kodi\addons\plugin.program.iarl\addon.py", line 1559, in download_and_launch_rom
                                                iarl_data['current_save_data'] = download_rom_only(iarl_data)
                                              File "C:\Users\pimve\AppData\Roaming\Kodi\addons\plugin.program.iarl\addon.py", line 1173, in download_rom_only
                                                iarl_data['current_save_data']['launch_filename'], post_download_action_success = post_download_action(iarl_data,iarl_data['current_rom_data']['rom_override_postdl'],None)
                                              File "C:\Users\pimve\AppData\Roaming\Kodi\addons\plugin.program.iarl\addon.py", line 1391, in post_download_action
                                                conversion_success, converted_filename = convert_zip_m3u(iarl_data)
                                              File "C:\Users\pimve\AppData\Roaming\Kodi\addons\plugin.program.iarl\resources\lib\util.py", line 2713, in convert_zip_m3u
                                                if '.cue' in [os.path.splitext(x)[-1] for x in os.listdir(unzip_folder_path)]:
                                            WindowsError: (3, 'Het systeem kan het opgegeven pad niet vinden', 'D:\\Games\\Retro Games\\PSX\\Spyro_Year_of_the_Dragon_USA_v11\\Spyro_the_Dragon_USA/*.*')
                                            -->End of Python script error report<--```

**Comment by zach-morris** [comment]
Thanks for the log.  I'll have to find a logical way to fix this, and first thought is I'll have to rely on 'best matching' somehow.

**Comment by zach-morris** [comment]
Fixed (I'm 99% sure) in v1.6.1

**Comment by PimmoZz** [comment]
Seems to be working indeed, good job! 👍
